distribution/push_v2.go

```diff
@@ -169,7 +170,13 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, ima
 
 	putOptions := []distribution.ManifestServiceOption{distribution.WithTag(ref.Tag())}
 	if _, err = manSvc.Put(ctx, manifest, putOptions...); err != nil {
-		logrus.Warnf("failed to upload schema2 manifest: %v - falling back to schema1", err)
+		logrus.Warnf("failed to upload schema2 manifest: %v", err)
+
+		if runtime.GOOS == "windows" {
+			return err
+		}
```

Comment:
Interesting. It makes sense that we would only want to push a schema2 manifest for windows, since it contains the base layer information, and a schema1 manifest omits that information. It will mean that Windows images require a sufficiently recent registry which supports the schema2 format. Do you see that being an issue at all? (cc @docker/core-distribution-maintainers)